src/test/java/de/mediathekview/mserver/crawler/sr/tasks/SrTopicsOverviewPageTaskTest.java

```diff
@@ -0,0 +1,55 @@
+package de.mediathekview.mserver.crawler.sr.tasks;
+
+import de.mediathekview.mserver.crawler.sr.SrTopicUrlDTO;
+import de.mediathekview.mserver.crawler.sr.SrConstants;
+import de.mediathekview.mserver.testhelper.JsoupMock;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import org.hamcrest.Matchers;
+import org.jsoup.Jsoup;
+import static org.junit.Assert.assertThat;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({Jsoup.class})
+public class SrTopicsOverviewPageTaskTest {
+  
+  private final SrTopicUrlDTO[] expectedUrls = new SrTopicUrlDTO[] {
+    new SrTopicUrlDTO("mag's", String.format(SrConstants.URL_SHOW_ARCHIVE_PAGE, "MA", 1)),
+    new SrTopicUrlDTO("Medienwelt", String.format(SrConstants.URL_SHOW_ARCHIVE_PAGE, "SR2_ME_P", 1)),
+    new SrTopicUrlDTO("Meine Traumreise", String.format(SrConstants.URL_SHOW_ARCHIVE_PAGE, "MT", 1)),
+    new SrTopicUrlDTO("mezz'ora italiana", String.format(SrConstants.URL_SHOW_ARCHIVE_PAGE, "AS_MEZI", 1)),
+    new SrTopicUrlDTO("Mit Herz am Herd", String.format(SrConstants.URL_SHOW_ARCHIVE_PAGE, "MHAH", 1)),
+    new SrTopicUrlDTO("MusikKompass", String.format(SrConstants.URL_SHOW_ARCHIVE_PAGE, "SR2_MK", 1)),
+    new SrTopicUrlDTO("MusikWelt", String.format(SrConstants.URL_SHOW_ARCHIVE_PAGE, "SR2_MUW", 1)),
+    new SrTopicUrlDTO("Nachrichten in einfacher Sprache", String.format(SrConstants.URL_SHOW_ARCHIVE_PAGE, "NIES_A", 1)),
+    new SrTopicUrlDTO("2 Mann für alle Gänge", String.format(SrConstants.URL_SHOW_ARCHIVE_PAGE, "ZMANN", 1))
+  };
+  
+  @Test
+  public void test() throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)